tests/integration/high_availability/test_replication.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+
+import lightkube
+import pytest
+import yaml
+from helpers import get_cluster_status, get_primary_unit
+from lightkube.resources.core_v1 import Pod
+from pytest_operator.plugin import OpsTest
+from tenacity import RetryError, Retrying, stop_after_delay, wait_fixed
+
+from tests.integration.high_availability.high_availability_helpers import (
+    get_max_written_value_in_database,
+)
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+MYSQL_APP_NAME = METADATA["name"]
+APPLICATION_APP_NAME = "application"
+CLUSTER_NAME = "test_cluster"
+TIMEOUT = 15 * 60
+
+
+@pytest.mark.order(1)
+@pytest.mark.abort_on_fail
+@pytest.mark.replication_tests
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build, deploy and relate the mysql and test applications."""
+    mysql_charm = await ops_test.build_charm(".")
```

Comment:
Take a peek on [this](https://github.com/canonical/mysql-operator/blob/fd6dbb6d75e29fcdbbb433510e4cf2b47b6df54c/tests/integration/test_ha.py#L36) - as commented bellow there's the necessity of enabling the tests to be run against an existing deployment.

Reply:
addressed in 0986dce467170ce6ac07013fda91adecffcf63ab - created a helper for set up where the above-like function is called for both mysql and the application